Shopify/dawn
# Add multirow section

### PR Summary: 

Added a new Multirow section for displaying related image and text content vertically.

### Why are these changes introduced?

Multirow is a section conceptually similar to `multicolumn`, but designed specifically for quicker and easier vertical content enumeration. Its blocks are aesthetically similar to the current `image with text` section.

Fixes #2169 

<img src="https://screenshot.click/13-35-t7p3r-f0s7q.png" width="350" />

### What approach did you take?

Copied the liquid from image with text into a new section, without creating a shared snippet, so much of the markup will be the same. I'm not certain the added snippet complexity that would be required to support both use cases would be worth it. This may be dependent on how each section functionally evolves in the future.

Multirow loops through blocks and creates a new image with text element for each, so the way existing content elements (heading, caption, etc) are read differs a bit from image with text. Certain image with text classes that were conditional on section settings were either removed or hardcoded if the settings are not offered in multirow.

### Other considerations

**Color scheme + global border settings logic** ([Video of behavior](https://screenshot.click/13-49-ajiff-u0u1z.mp4))
I've included the same "magic" as image with text where if the container color scheme chosen matches the section background and _no_ borders or shadow styles are applied via global content container settings. This allows either a [unified container look](https://screenshot.click/13-49-qnvpx-flt77.png) or a [container-less/transparent look](https://screenshot.click/13-48-ncbh5-h0lu0.png). The only difference here is that multirow also allows the option to change the section background so this logic isn't solely linked to background-1

**Shadow stacking limitations** ([Video of behavior](https://screenshot.click/13-30-dsr8l-av93i.mp4))
When vertical shadow offsets for either the global media or content container settings are negative (upward), the shadow could overlap the row above (natural CSS behavior) if there isn't enough gap. The issue here is there are 2 different child elements within an image with text creating their own shadows with an existing stacking relationship of their own, which further adds to the complexity that we've been able to solve for in other sections. I briefly timeboxed a fix for this and didn't find a configuration that worked, but I don't think it's blocking.

**Color scheme + padding logic** ([Image of behavior](https://screenshot.click/03-18-fewou-8xi43.png))
We also aligned to include additional logic to better align the text content with the standard page margins when both color scheme settings are the same and the rows don't create the "container" look. Basically it just removes the appropriate left or right padding depending on the row's content alignment. An exception is also made for centered text.

### Decision log

| # | Decision | Alternatives | Rationale | Downsides |
|---|---|---|---|---|
| 1 | Each block is a row, rather than a single content element | Basically none | - To allow for multiple rows<br>- To allow for connection to dynamic source groups | - Content elements will not be reorder-able like image with text |
| 2 | Block content style settings at the section level | Include all/more content style settings at the block level | - Focus is around quick and easy configurable<br>- More granularity at the block level would be more tedious and time consuming | - All blocks will display with the same setting choices |
| 3 | Did not include _all_ currently available settings from image with text | Include more or all image with text settings | - Focus on settings most relevant to the purpose of this section<br>- Less options, less configuration time for merchants<br>- Easier to add settings later than take some away | - Rows will not have all the same styling options as image with text (e.g overlap layout) |
| 4 | Did not reuse existing translated strings from image with text | Reuse common strings and only add multirow strings when needed | - Allows image with text and multirow to more easily diverge further in the future<br>- Keeps things simpler | - Inherent redundancy |
| 5 | Did not abstract image with text liquid into a reusable snippet | Consolidate core image with text in a settings-agnostic snippet | - Allows image with text and multirow to more easily diverge further in the future<br>- Snippet would need to be completely agnostic of `section.settings` and `block.settings` references resulting in a lot of props<br>- New logic specific to one component will require adding an exception in the shared snippet<br>- I'm not confident if shared liquid here will be beneficial or a burden | - Inherent redundancy<br>- New shared logic will need to be duplicated  |

### Visual impact on existing themes
Multirow is a new section and any changes to image with text styles should be scoped to the multirow context, with one minor exception.. Long words that might not have broken in image with text captions or buttons before should break now ([example](https://camo.githubusercontent.com/72c9f691fc93175f5fc095db289d1f17c17e80672b82fb1ece00ec306309b85a/68747470733a2f2f73637265656e73686f742e636c69636b2f31302d35312d656c646f6d2d6a357767352e706e67))

### Testing steps/scenarios
- Test Multirow sections with various combinations of section settings
- Ensure all settings/options are labeled correctly and behave as expected
- Test color scheme combinations thoroughly
- Test with different combination of global media and content container settings
- Test with dynamic content sources (Puppy product)

~~Note: `Image height: medium` setting may not apply correctly until the PR that adds the styles for this is merged and this one is rebased.~~

### Demo links

- [Editor](https://os2-demo.myshopify.com/admin/themes/139164581910/editor)

### Checklist
